includes/core-changes/serialization/5.0/jsonserializer-allows-reading-numbers-as-strings.md

```diff
@@ -0,0 +1,53 @@
+### ASP.NET Core apps allow deserializing quoted numbers
+
+Starting in .NET 5.0, ASP.NET Core apps use the default deserialization options as specified by <xref:System.Text.Json.JsonSerializerDefaults.Web?displayProperty=nameWithType>. The <xref:System.Text.Json.JsonSerializerDefaults.Web> set of options includes setting <xref:System.Text.Json.JsonSerializerOptions.NumberHandling> to <xref:System.Text.Json.Serialization.JsonNumberHandling.AllowReadingFromString?displayProperty=nameWithType>. This change means that ASP.NET Core apps will successfully deserialize numbers that are represented as JSON strings, instead of throwing an exception.
+
+#### Change description
+
+In .NET Core 3.0 - 3.1, <xref:System.Text.Json.JsonSerializer> throws a <xref:System.Text.Json.JsonException> during deserialization if it encounters a quoted number in a JSON payload. The quoted numbers are used to map with number properties in object graphs. In .NET Core 3.0 - 3.1, numbers are only read from <xref:System.Text.Json.JsonTokenType.Number?displayProperty=nameWithType> tokens.
+
+Starting in .NET 5.0, quoted numbers in JSON payloads are considered valid, by default, for ASP.NET Core apps. No exception is thrown during deserialization of quoted numbers.
+
+> [!TIP]
+>
+> - There is no behavior change for the default, standalone <xref:System.Text.Json.JsonSerializer> or <xref:System.Text.Json.JsonSerializerOptions>.
+> - This is technically not a breaking change, since it makes a scenario more permissive instead of more restrictive (that is, it succeeds in coercing a number from a JSON string instead of throwing an exception). However, since this is a significant behavioral change that affects many ASP.NET Core apps, it is documented here.
+> - The <xref:System.Net.Http.Json.HttpClientJsonExtensions.GetFromJsonAsync%2A> and <xref:System.Net.Http.Json.HttpContentJsonExtensions.ReadFromJsonAsync%2A> extension methods also use the <xref:System.Text.Json.JsonSerializerDefaults.Web> set of serialization options.
```

Comment:
Given `GetFromJsonAsync` and `ReadFromJsonAsync` are from a different namespace and don't ship with the `System.Text.Json` NuGet package, it might be better to highlight these more, or maybe a separate thing.

At  minimum, can we make the whole typename with the namespace visible?
![image](https://user-images.githubusercontent.com/6527137/96636126-b0497100-12d1-11eb-8f8c-84bb9bb012d0.png)
```suggestion
	> - The <xref:System.Net.Http.Json.HttpClientJsonExtensions.GetFromJsonAsync%2A?displayProperty=nameWithType> and <xref:System.Net.Http.Json.HttpContentJsonExtensions.ReadFromJsonAsync%2A?displayProperty=nameWithType> extension methods also use the <xref:System.Text.Json.JsonSerializerDefaults.Web> set of serialization options.
```

OR
Make sure to call out explicitly in the samples/docs that the default behavior is different. I see, we have it as a remark, but stating explicitly that it uses Web, and JsonSerializer on its own doesn't is a good distinction to make:
https://review.docs.microsoft.com/en-us/dotnet/api/system.net.http.json.httpclientjsonextensions.getfromjsonasync?view=net-5.0&branch=master#System_Net_Http_Json_HttpClientJsonExtensions_GetFromJsonAsync_System_Net_Http_HttpClient_System_String_System_Type_System_Threading_CancellationToken_

Reply:
> .GetFromJsonAsync%2A>
Unrelated question, what's the %2A for? An asterisk `*`?

Reply:
Yes, %2A is the asterisk, refers to all overloads, renders without () after the method name.

Reply:
I updated them [here](https://github.com/dotnet/dotnet-api-docs/blob/29963953d45ed3ef3fbd848246db96f2e79c817f/xml/System.Net.Http.Json/HttpClientJsonExtensions.xml#L53). Does it look okay?

Reply:
Yes, it might be easier to review in a (draft or regular) PR.

Regarding this commit:
https://github.com/dotnet/dotnet-api-docs/commit/29963953d45ed3ef3fbd848246db96f2e79c817f
And remarks like:
https://github.com/dotnet/dotnet-api-docs/blob/29963953d45ed3ef3fbd848246db96f2e79c817f/xml/System.Net.Http.Json/HttpClientJsonExtensions.xml#L346

When it says `serialization` for one, the verb needs to be the same for the other.

Right now, all the additions say deserialization: `whereas <xref:System.Text.Json.JsonSerializer> deserialization methods do not, by default.`

Reply:
Ah, thanks for catching that. I've updated the verbs to match as part of this PR: https://github.com/dotnet/dotnet-api-docs/pull/5007.